setup.py

```diff
@@ -110,6 +110,7 @@ def get_r_c_extension_status():
                                     library_dirs=c_ext.library_dirs)
     return status
 
+dist.Distribution().fetch_build_eggs(["packaging;platform_system=='Windows'"])
```

Comment:
Is this necessary? I am not very familiar with conditional rows in `requirements.txt` but the doc suggests that adding `platform_system=='Windows'` will take care of it.

Reply:
It is required to run `setup.py` (with the version resolution). Without this line, `pip` fails to run as `setup.py` indirectly `import packaging` as it calls `situation.get_r_home()` before `setuptools.setup()` is called (to process `setup_requires` option).

One way to simplify all this is to just use the last subkey of `HKEY_LOCAL_MACHINE\Software\R-core\R` and skip the version check. As long as R version doesn't go up to 2 digits, this should be fine I think. (i.e., if a hypothetical system has both R v4.9.0 and v4.10.0 it may pick v4.9.0 instead.)

Picking the last registry entry will remove the dependence on `packaging` package thus eliminates all the little changes elsewhere you commented here and above. I coded up the most accurate implementation and not partial to either approaches.

Reply:
Thanks. Python package is definitely not without intricacies. I am catching up with you.

I am trying to understand how the top-level call `dist.Distribution().fetch_build_eggs()` is needed.
I am reading that `setup()` has an argument `setup_requires` (which would be the more canonical way to achieve this). However, I am also reading in PEP0518 that this no longer recommended (https://peps.python.org/pep-0518/).

I addition to that I am realizing that I do not know where the dependency on `packaging` is coming from and why it would be Windows-specific.

Reply:
Hey, I'm on the same boat. I'm on constant learning mode for Python.

> where the dependency on packaging is coming from and why it would be Windows-specific.

`packaging` is what I added in my PR to run the version check in `r_home_from_registry()` and `fetch_build_eggs(...)` is needed because this part of `setup.py` runs before `setuptools`' require logic kicks in. Specifically, `set_r_c_extension_status()` implicitly calls `r_home_from_registry()` right after this line.

>I am reading that setup() has an argument setup_requires

Tried that already :-D But it didn't help the situation because of the reason stated above: `setup.py` uses `packaging` module before `setup_requires` is being processed. As much as I hated to insert this `dist.Distribution().fetch_build_eggs()` line, I couldn't figure out a better way without making too many changes around the package setup scripts.

If you are OK for me to make more drastic changes to the package setup configuration, like making it up to date with PEP518 to modernize the setup script, I'd be happy to work on that (perhaps as a separate PR). I have a couple ideas to clean it up.